pkg/kubelet/lifecycle/predicate.go

```diff
@@ -124,24 +140,11 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 			}
 		}
 		// If there are failed predicates, we only return the first one as a reason.
-		r := reasons[0]
-		switch re := r.(type) {
-		case *predicates.PredicateFailureError:
-			reason = re.PredicateName
-			message = re.Error()
-			klog.V(2).Infof("Predicate failed on Pod: %v, for reason: %v", format.Pod(admitPod), message)
-		case *predicates.InsufficientResourceError:
-			reason = fmt.Sprintf("OutOf%s", re.ResourceName)
-			message = re.Error()
-			klog.V(2).Infof("Predicate failed on Pod: %v, for reason: %v", format.Pod(admitPod), message)
-		default:
-			reason = "UnexpectedPredicateFailureType"
-			message = fmt.Sprintf("GeneralPredicates failed due to %v, which is unexpected.", r)
-			klog.Warningf("Failed to admit pod %v - %s", format.Pod(admitPod), message)
-		}
+		message = statuses.Merge().Message()
+		klog.V(2).Infof("Predicate failed on Pod: %v, for reason: %v", format.Pod(admitPod), message)
 		return PodAdmitResult{
 			Admit:   fit,
-			Reason:  reason,
+			Reason:  "PredicateFailure",
```

Comment:
I suppose this will break some tests? As in the before, it can be something like "OutOfMemroy", but it's not right now.